kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/MetricsIT.java

```diff
@@ -454,6 +568,173 @@ void shouldIncrementConnectionMetrics(KafkaCluster cluster, Topic topic) throws
+                                            SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, clientTrustStore.toAbsolutePath().toString(),
+                                            SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, downstreamCertificateGenerator.getPassword()))) {
+                                assertThat(admin.describeCluster(failQuickDescribeCluster).clusterId())
+                                        .failsWithin(Duration.ofSeconds(5))
+                                        .withThrowableThat()
+                                        .withCauseInstanceOf(TimeoutException.class)
+                                        .havingCause()
+                                        .withMessageStartingWith("Timed out waiting for a node assignment.");
+                            }
+                        }));
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "createDownstreamConnectionError")
+    void shouldIncrementDownstreamErrorOnConnectionError(Supplier<VirtualClusterGateway> gatewaySupplier, Consumer<KroxyliciousTester> causeConnectionError,
+                                                         KafkaCluster cluster) {
+        var config = proxy(cluster)
+                .withNewManagement()
+                .withNewEndpoints()
+                .withNewPrometheus()
+                .endPrometheus()
+                .endEndpoints()
+                .endManagement()
+                .editFirstVirtualCluster()
+                .withGateways(List.of(gatewaySupplier.get()))
+                .endVirtualCluster();
+
+        // Given
+        try (var tester = kroxyliciousTester(config);
+                var managementClient = tester.getManagementClient()) {
+
+            // When
+            causeConnectionError.accept(tester);
+
+            // Then
+            await().atMost(Duration.ofSeconds(30))
+                    .untilAsserted(() -> {
+                        var updatedMetricsList = managementClient.scrapeMetrics();
+                        assertMetricsWithValue(updatedMetricsList, "kroxylicious_client_to_proxy_errors_total", null);
+                    });
+        }
+    }
+
+    @Test
+    void shouldIncrementUpstreamErrorOnConnectionError(@Tls KafkaCluster cluster) {
+        var brokerTruststore = (String) cluster.getKafkaClientConfiguration().get(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG);
+        var brokerTruststorePassword = (String) cluster.getKafkaClientConfiguration().get(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG);
+        assertThat(brokerTruststore).isNotEmpty();
+        assertThat(brokerTruststorePassword).isNotEmpty();
+
+        var builder = new ConfigurationBuilder()
+                .addToVirtualClusters(new VirtualClusterBuilder()
+                        .withName("demo")
+                        .withNewTargetCluster()
+                        .withBootstrapServers("Wrong:9092")
```

Comment:
be wary of relying on names that don't resolve.  Sometimes you might find a test environment where the test starts hanging until DNS times-out.

In this case... what I'd do it use wire the Proxy up to the Kafka Cluster as normal.
Then in your test, call `cluster.stop()`;  That will guarantee that the Broker won't pick up the phone when the Proxy calls. Viola.   That's a simple way to produce the failing case you want with minimum code.